pageserver/src/tenant.rs

```diff
@@ -1399,16 +1462,41 @@ impl Tenant {
     }
 
     pub(crate) async fn apply_timeline_archival_config(
-        &self,
+        self: &Arc<Self>,
         timeline_id: TimelineId,
         state: TimelineArchivalState,
```

Comment:
nit: it wasn't added in this PR, but `state` is a bit confusing as a name when reading the function, because it's not clear if it means the current state or the intended state.  Maybe something like `new_state`?

Reply:
done